sdks/python/apache_beam/transforms/util_test.py

```diff
@@ -547,7 +547,9 @@ def test_in_streaming_mode(self):
                    .advance_watermark_to(start_time +
                                          GroupIntoBatchesTest.NUM_ELEMENTS)
                    .advance_watermark_to_infinity())
-    pipeline = TestPipeline()
+    options = PipelineOptions()
+    options.view_as(StandardOptions).streaming = True
+    pipeline = TestPipeline(options=options)
```

Comment:
Here you can just write `pipeline = TestPipeline(options= StandardOptions(streaming=True))`

Reply:
There have been a few tests that use TestStream without advancing to infinity at the end. My understanding is that they were only ever run on the direct runner, so maybe it's not strictly required for direct runner?

Reply:
By definition all watermarks advance to infinity implicitly when there is no more input data within a pipeline.

We could:
* make advance to infinity explicitly required in all test streams implementations/usages (Java does this and what this CL aims to do in Python).
* make advance to infinity implicit in all test streams implementations and fix issues with runners to make sure they actually advance to infinity to fix the issues (Python may have been doing this but direct runner may have had bugs).

I'm for the explicit route since test writers should understand that their output validation should still handle the watermark advancement to infinity.

Reply:
I'm also in favor of explicitly advancing the watermark.